File: kafka-helper-domain/src/main/java/com/hg/kafka/helper/domain/entity/HashkeyInfo.java
package com.hg.kafka.helper.domain.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;

@Document(collection = "hashkey_info")
@Data
public class HashkeyInfo {
    @Id
    private String id;
    @Field("hashkey")
    private String hashkey;
    @Field("name")
    private String name;
    @Field("topic")
    private String topic;
    @Field("desc")
    private String description;
    @Field("group_id")
    private String groupId;
    @Field("created_at")
    private Date createdAt;
    @Field("updated_at")
    private Date updatedAt;
}

File: kafka-helper-domain/src/main/kotlin/com/hg/kafka/helper/domain/entity/HashkeyInfo.kt
package com.hg.kafka.helper.domain.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.mapping.Field
import java.util.*

@Document("hashkey_info")
data class HashkeyInfo(
        @Id var id: String? = null,
        @Field("hashkey") var hashkey: String,
        @Field("name") var name: String,
        @Field("topic") var topic: String,
        @Field("desc") var description: String,
        @Field("group_id") var groupId: String,
        @Field("created_at") var createdAt: Date,
        @Field("updated_at") var updatedAt: Date
)